### PR TITLE
Make activation polling for blocking invocations configurable

### PR DESCRIPTION
**Pull request by dubee**
## Description

Allows controller activation polling for blocking invocations to be enabled or disable via configuration.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [x] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by dubee** [comment]
Changed this back to a WIP state as the two active ack changes do not currently allow DB polling to be disabled.

File: core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala [hunk 1]
```diff
@@ -593,11 +596,15 @@ protected[actions] trait PrimitiveActions {
     //    in case of an incomplete active-ack (record too large for example).
     activeAckResponse.foreach {
       case Right(activation) => result.trySuccess(Right(activation))
-      case _                 => pollActivation(docid, context, result, i => 1.seconds + (2.seconds * i), maxRetries = 4)
+      case _ if (controllerActivationConfig.pollingFromDb) =>
+        pollActivation(docid, context, result, i => 1.seconds + (2.seconds * i), maxRetries = 4)
```
Comment (tysonnorris) [inline core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala]: I'm looking at a case right now where if this pollActivation is removed, the acks are received in the wrong order in some cases (completion before result) - have you seen anything similar? unfortunately, it is something weird where it only happens on first run of the controller - subsequent runs of same or different actions don't exhibit the problem.  Restarting controller, and the first invocation always gets the acks in wrong order again.

Comment (tysonnorris) [reply core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala]: I think I see the issue causing the out of order messages - the first call to `AcknowledegmentMessage.parse(raw)` is slow, resulting in out of order messages IFF the result + completion are sent at approx same time from invoker. I'll try to figure out a way to force the order of these.

Comment (markusthoemmes) [reply core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala]: @tysonnorris is this related to #4134, which supposedly has been fixed by now?

Comment (dubee) [reply core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala]: After rebasing it looks the two active ack changes no longer required DB polling.

**Comment by dubee** [comment]
@rabbah, rebased and ready to go.

**Pull request merged**